### PR TITLE
Add claims pallet

### LINKED ISSUE #76: add and test claim pallet
In case we want to allow claims from ethereum addresesses, we need a way to verify ETH signatures to transfer allowances to TEER accounts

Polkadot implemented such a pallet already:
https://github.com/paritytech/polkadot/blob/master/runtime/common/src/claims.rs

As we don't need all the `common` crate, we should just copy that pallet code to our repo (must keep GPLv3!) as its own crate
https://github.com/integritee-network/pallets

Test happy flow:
1. create Ethereum address with metamask
2. `Claims::mint_claim()` to set allowance for the above generated address. Add a vesting schedule
3. create a integritee address with polkadot extension
4. sign a message with the ETH address in metamsk with a payload as indicated in the `claim`code
5. `Claims::claim()`
6. check balance and vesting schedule of generated integritee address

Test adversary scenario:
* try to claim with the right ETH pubkey but wrong signature. must fail

### PR DESCRIPTION
**Pull request by haerdib**
Test happy flow:

- [x]     create Ethereum address with metamask: 0x86Ad0947B3aF70A369D865FAfe4386ecB177c828
- [x]     Claims::mint_claim() to set allowance for the above generated address. Add a vesting schedule
- [x]     create a integritee address with polkadot extension
- [x]     sign a message with the ETH address in metamsk with a payload as indicated in the claimcode
- [x]     Claims::claim()
- [x]     check balance and vesting schedule of generated integritee address

Test adversary scenario:   

- [x] try to claim with the right ETH pubkey but wrong signature. must fail
--------------
Necessary code updates: 
- [x] merge & change dependencies to master https://github.com/integritee-network/pallets/pull/2

closes #76

### CODE REVIEW

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -375,6 +375,7 @@ fn genesis_config(
 			key: root_key,
 		},
 		teerex: TeerexConfig { allow_sgx_debug_mode: true },
+		claims: Default::default(),
```
Comment (haerdib) [inline node/src/chain_spec.rs:378]: @brenzi can you agree with that or do you want to add a specifc genesis config?

Comment (brenzi) [reply node/src/chain_spec.rs:378]: that's fine. we can populate claims on the fly

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -377,6 +379,19 @@ impl pallet_teerex::Config for Runtime {
...
+parameter_types! {
+	pub Prefix: &'static [u8] = b"Pay TEERs to the integriTEE account:";
+}
+
+/// added by Integritee
+impl pallet_claims::Config for Runtime {
+	type Event = Event;
+	type VestingSchedule = Vesting;
+	type Prefix = Prefix;
+	type MoveClaimOrigin = frame_system::EnsureRoot<AccountId>;
```
Comment (haerdib) [inline runtime/src/lib.rs:391]: I've set that to Root only to keep it simple. Polkadot is using a majority vote of the council

**Review by brenzi — APPROVED** [review]
LGTM

**Comment by haerdib** [comment]
Minting and claiming is working. There are some things I'm unsure about:

That should be the expected output of a claim with +1000 and a vesting schedule of (<BalanceOf>500, <BalanceOf>100, <BlockNr>300) I guess:
grafik

So I was able to successfully claim via developer -> extrinsic.

But Accounts -> Claim Tokens did not work because I was not able to press "Confirm claim" 
grafik
 
 No matter what I entered.. is that an expected behaviour?

**Comment by haerdib** [comment]
So, I've found the issue. I was finally able to claim it: 
grafik

The problem was that MyEtherWallet (which I used for signing) was giving back the following signature:
```
{
  "address": "0x86ad0947b3af70a369d865fafe4386ecb177c828",
  "msg": "0x50617920544545527320746f2074686520696e7465677269544545206163636f756e743a64363037303563393563313065336265393435363438663861623664366336613962336230396462343065383361386136313065306563393134633535633761",
  "sig": "3603071c9638731ccad978dcef7adc90030d687cb128dc696a6d606aeedd77da39ff1ebd2a03b7432e8d3682adcbd3b9ebd5f73d7dddf3f126b473b49dee83761b",
  "version": "3",
  "signer": "MEW"
}
```
But polkadot expected a signature starting with `0x`, which I had to include manually. Another interesting thing: When I entered the correct signature blob, the polkadot portal automatically refilled the ETH Adress field with a different one - stating that there was no claim for the entered ETH address.. which ofc is true because the automatically entered one was not correct. It only worked after I reentered my ETH address.

**Comment by haerdib** [comment]
Other than the mentioned issue above this PR is ready for merge.

**Pull request merged**